### PR TITLE
Add reference fact recall documentation

### PR DESCRIPTION
**Pull request by gakhromov**
Adds documentation on our new Reference Fact Recall metric.

### CODE REVIEW

**Review by miivanov90 — APPROVED** [review]
Great, thanks!

File: docs/foresight/metrics/reference_fact_recall.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- An LLM's generated response to be evaluated;
+- A reference response to compare the generated response with.
+
+This metric checks that the answer given by the LLM is mentioning all the facts
+listed in the reference answer. Additional information is not penalised.
+
+Example:
+- **Question**: *Give me a checklist to prepare for my hiking trip to the mountains.*
+- **Reference response**: *You should bring your a water bottle, hiking shoes and sunscreen.* 
+- **Candidate answer 1**: *Here is a list of items to bring: 1) hiking shoes; 2) a water bottle.* `[reference fact recall score = 0.75]`
```
Comment (apanakkat) [inline docs/foresight/metrics/reference_fact_recall.md:19]: Shouldn't the score here be 0.67?

2/3 facts?

**Review by apanakkat — COMMENTED** [review]
Thanks G.

File: docs/foresight/metrics/reference_fact_recall.md (hunk 1 above)
Comment (gakhromov) [reply docs/foresight/metrics/reference_fact_recall.md:19]: You are right, haha. I think I removed one fact to make the example easier to parse but forgot to change the score...

**Pull request merged**